src/physics/zenith.jl

```diff
@@ -0,0 +1,283 @@
+) where NF
+
+    (;sinlat, coslat, lons) = geometry
+    (;cos_zenith, length_of_day, length_of_year) = S
+
+    # g: angular fraction of year [0...2π] for Jan-01 to Dec-31
+    time_of_year = S.seasonal_cycle ? time : S.initial_time[]
+    g = year_angle(NF, time_of_year, length_of_day, length_of_year)
+
+    # time correction [radians] due to the equation of time (sunrise/set oscillation)
+    tc = S.equation_of_time ? S.time_correction(g) : 0
+    
+    # solar hour angle at 0˚E (longtiude offset added later)
+    λ = 0
+    solar_hour_angle_0E = solar_hour_angle(NF, time, λ, length_of_day) + tc
+
+    # solar declination angle [radians] changing from tropic of cancer to capricorn
+    # throughout the year measured by g [radians]
+    δ = S.solar_declination(g)
+    sinδ,cosδ = sincos(δ)
+
+    rings = eachring(cos_zenith)
+
+    @inbounds for (j,ring) in enumerate(rings)                         
+        sinδsinϕ = sinδ*sinlat[j]
+        cosδcosϕ = cosδ*coslat[j]
+        for ij in ring
+            h = solar_hour_angle_0E + lons[ij]      # solar hour angle at longitude λ in radians
+            cos_zenith[ij] = max(0,sinδsinϕ + cosδcosϕ*cos(h))
+        end
+    end
+end
+
+"""Solar zenith angle varying with seasonal cycle only.
+$(TYPEDFIELDS)"""
+Base.@kwdef struct SolarZenithSeason{NF<:AbstractFloat,Grid<:AbstractGrid{NF}} <: AbstractZenith{NF,Grid}
+    # DIMENSIONS
+    nlat_half::Int
+
+    # OPTIONS
+    length_of_day::Second = Hour(24)
+    length_of_year::Second = Day(365.25)
+    seasonal_cycle::Bool = true
+
+    # COEFFICIENTS
+    solar_declination::SinSolarDeclination{NF} = SinSolarDeclination{NF}()
+
+    # WORK ARRAYS
+    initial_time::Base.RefValue{DateTime} = Ref(DEFAULT_DATE)
+    cos_zenith::Grid = zeros(Grid,nlat_half)
+end
+
+function cos_zenith!(
+    S::SolarZenithSeason{NF},
+    time::DateTime,
+    geometry::Geometry
+) where NF
+
+    (;sinlat, coslat, lat) = geometry
+    (;cos_zenith, length_of_day, length_of_year) = S
+
+    # g: angular fraction of year [0...2π] for Jan-01 to Dec-31
+    time_of_year = S.seasonal_cycle ? time : S.initial_time[]
+    g = year_angle(NF, time_of_year, length_of_day, length_of_year)
+    
+    # solar declination angle [radians] changing from tropic of cancer to capricorn
+    # throughout the year measured by g [radians]
+    δ = S.solar_declination(g)
+    sinδ,cosδ = sincos(δ)
+
+    local h₀::NF                # hour angle sunrise to sunset
+    local cos_zenith_j::NF      # at latitude j
+
+    rings = eachring(cos_zenith)
+    @inbounds for (j,ring) in enumerate(rings)
+            
+        ϕ = lat[j]
+        h₀ = abs(δ) + abs(ϕ) < π/2 ?        # polar day/night?
+        acos(-tan(ϕ) * tan(δ)) :            # if not: calculate length of day
+        ϕ*δ > 0 ? π : 0                     # polar day if signs are equal, otherwise polar night
+        
+        sinϕ, cosϕ = sinlat[j], coslat[j]
+        cos_zenith_j = max(0,h₀*sinδ*sinϕ + cosδ*cosϕ*sin(h₀))
```

Comment:
```suggestion
        cos_zenith_j = h₀*sinδ*sinϕ + cosδ*cosϕ*sin(h₀)
```
probably don't need a max here